crates/ruff_annotate_snippets/src/renderer/display_list.rs

```diff
@@ -209,18 +214,51 @@ impl DisplaySet<'_> {
         if formatted_len == 0 {
             self.format_label(line_offset, &annotation.label, stylesheet, buffer)
         } else {
-            let id = match &annotation.id {
-                Some(id) => format!("[{id}]"),
-                None => String::new(),
-            };
-            buffer.append(
-                line_offset,
-                &format!("{}{}", annotation_type_str(&annotation.annotation_type), id),
-                *color,
-            );
+            // TODO(brent) All of this complicated checking of `hide_severity` should be reverted
```

Comment:
Nice comment! This helped me a lot to understand what's going on. Thanks for writing it